isis/tests/SpiceRotationTests.cpp

```diff
@@ -0,0 +1,822 @@
+  rot.SetPolynomial(angle1Coeffs, angle2Coeffs, angle3Coeffs, SpiceRotation::PolyFunctionOverSpice);
+
+  rot.SetEphemerisTime(0.0);
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, rot.Matrix(),
+                      (vector<double>{0.0, 1.0, 0.0,
+                                     -1.0, 0.0, 0.0,
+                                      0.0, 0.0, 1.0}),
+                      testTolerance);
+
+  rot.SetEphemerisTime(1.0);
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, rot.Matrix(),
+                      (vector<double>{0.0, 1.0, 0.0,
+                                     -1.0, 0.0, 0.0,
+                                      0.0, 0.0, 1.0}),
+                      testTolerance);
+
+  rot.SetEphemerisTime(2.0);
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, rot.Matrix(),
+                      (vector<double>{-1.0, 0.0,  0.0,
+                                       0.0, 1.0,  0.0,
+                                       0.0, 0.0, -1.0}),
+                      testTolerance);
+
+  rot.SetEphemerisTime(3.0);
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, rot.Matrix(),
+                      (vector<double>{0.0,  0.0, -1.0,
+                                      1.0,  0.0,  0.0,
+                                      0.0, -1.0,  0.0}),
+                      testTolerance);
+
+
+  // Convert to a cache and ensure it still matches
+}
+
+
+TEST_F(SpiceRotationIsd, VectorRotation) {
+  SpiceRotation rot(-94031);
+  rot.LoadCache(isd);
+
+  vector<double> unitX = {1.0, 0.0, 0.0};
+  vector<double> unitY = {0.0, 1.0, 0.0};
+  vector<double> unitZ = {0.0, 0.0, 1.0};
+
+  rot.SetEphemerisTime(1.0);
+
+  EXPECT_PRED_FORMAT3(AssertVectorsNear,
+                      rot.J2000Vector(unitX),
+                      unitY, testTolerance);
+
+  EXPECT_PRED_FORMAT3(AssertVectorsNear,
+                      rot.J2000Vector(unitY),
+                      (vector<double>{-1.0, 0.0, 0.0}), testTolerance);
+
+  EXPECT_PRED_FORMAT3(AssertVectorsNear,
+                      rot.J2000Vector(unitZ),
+                      unitZ, testTolerance);
+
+  EXPECT_PRED_FORMAT3(AssertVectorsNear,
+                      rot.ReferenceVector(unitX),
+                      (vector<double>{0.0, -1.0, 0.0}), testTolerance);
+
+  EXPECT_PRED_FORMAT3(AssertVectorsNear,
+                      rot.ReferenceVector(unitY),
+                      unitX, testTolerance);
+
+  EXPECT_PRED_FORMAT3(AssertVectorsNear,
+                      rot.ReferenceVector(unitZ),
+                      unitZ, testTolerance);
+}
+
+
+TEST_F(SpiceRotationIsd, PolynomialPartials) {
+  SpiceRotation rot(-94031);
+  rot.LoadCache(isd);
+  rot.ComputeBaseTime();
+  rot.SetPolynomialDegree(1);
+  // The base time is set to 1.5, and the time scale is set to 1.5 so these
+  // coefficients are scaled accordingly. The unscaled equations are:
+  // angle1 = -pi/2 + pi/2 * t
+  // angle2 = -pi   + pi/2 * t
+  // angle3 =  pi   - pi/2 * t
+  //
+  // Note ISIS defaults to ZXZ rotation axis order
```

Comment:
Should the comment say ZXZ or should a Y be included?

Reply:
It should be ZXZ